### PR TITLE
[two_dimensional_scrollables] Fix layout offset of merged pinned cells

### PR DESCRIPTION
**Pull request by Piinks**
Fixes https://github.com/flutter/flutter/issues/143526

This fixes the layout offset computation when there are merged cells within pinned rows and/or columns in TableView.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by Piinks** [comment]
Just when I thought https://github.com/flutter/packages/pull/5917 had all the tests! 🤣

File: packages/two_dimensional_scrollables/test/table_view/table_test.dart [hunk 1]
```diff
@@ -1730,6 +1730,478 @@ void main() {
...
+                rowMergeStart:
+                    bothMerged[vicinity]?.start ?? rowMerged[vicinity]?.start,
+                rowMergeSpan:
+                    bothMerged[vicinity]?.span ?? rowMerged[vicinity]?.span,
+                child: Text(
+                  'R${bothMerged[vicinity]?.start ?? rowMerged[vicinity]?.start ?? vicinity.row}:'
+                  'C${bothMerged[vicinity]?.start ?? columnMerged[vicinity]?.start ?? vicinity.column}',
+                ),
+              );
+            });
```
Comment (goderbauer) [inline packages/two_dimensional_scrollables/test/table_view/table_test.dart:1796]: nitty-nit: trailing comma here for slightly nicer formatting with less indentation?

```suggestion
            },);
```

Comment (goderbauer) [reply packages/two_dimensional_scrollables/test/table_view/table_test.dart:1796]: same elsewhere in this file.

**Review by goderbauer — APPROVED** [review]
LGTM!

**Pull request merged**